rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+      parameters:
+        - name: namespace
+          description: Namespace under which to list tables.
+          in: path
+          required: true
+          schema:
+            type: string
+          examples:
+            singlepart_namespace:
+              value: "accounting"
+            multipart_namespace:
+              value: "accounting.tax"
+        # TODO - There's a much more native way to handle pagination
+        - name: limit
+          description: number of values to return in one request
+          in: query
+          required: false
+          schema:
+            type: integer
+          example: 10
+        - name: offset
+          description: Place in the response to continue from if paginating
+          in: query
+          required: false
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListTablesResponse'
+    post:
+      tags:
+        - Catalog API
+      summary: Create a table with the identifier given in the body
+      operationId: createTable
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateTableRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CreateTableResponse'
+    put:
+      tags:
+        - Catalog API
+      summary: Rename a table from its current name to a new name within the same catalog
+      description: Rename a table within the same catalog
+      operationId: renameTable
+      requestBody:
+        description: Current table identifier to rename and new table identifier to rename to
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RenameTableRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Unauthorized
+        # Not using 404 as 404 is very hard to monitor.
+        # There's no way to tell by monitoring returned response codes
+        # if a client made a valid request but the table didn't exist,
+        # or if a client or server has been misconfigured or some bug exists
+        # and the client is actually calling endpoints that the server doesn't
+        # have.
+        #
+        # We can consider a different response code to use than 412.
+        '412':
+          description: Table to rename from does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchTableError'
+              example:
+                '{ error: { message: "Table does not exist", type: "NoSuchTableException", code: 41202 }'
```

Comment:
Looks like the example code is 5 digits in a few places.

Reply:
Yeah the example code is 5 digits.

The first 3 digits are the HTTP response code, the second two digits are defined by us.

In this example, the code is 41202: 412 being the HTTP response code, and 02 meaning "table does not exist". My thinking was just that we'd have a set of codes in the spec that we follow that one could easily look up to get more information on the cause of the failure.

I'm open to removing that field entirely though if we'd like (or just sticking the HTTP response code in the error body).